File: start-dev.py
#!/usr/bin/env python3
"""
Development startup script for Curamete application.
This script starts both the frontend and backend servers.
"""

import subprocess
import sys
import os
import time
from threading import Thread

def start_frontend():
    """Start the frontend development server"""
    print("Starting frontend server...")
    try:
        # Use shell=True to ensure the command runs properly on Windows
        frontend = subprocess.Popen(
            "npm run dev",
            shell=True,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            text=True
        )
        
        # Print frontend output
        for line in frontend.stdout:
            print(f"[FRONTEND] {line.strip()}")
            
    except Exception as e:
        print(f"Error starting frontend: {e}")

def start_backend():
    """Start the backend server"""
    print("Starting backend server...")
    try:
        # Change to backend directory
        backend_dir = os.path.join(os.getcwd(), "backend")
        os.chdir(backend_dir)
        
        backend = subprocess.Popen(
            "python main.py",
            shell=True,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            text=True
        )
        
        # Print backend output
        for line in backend.stdout:
            print(f"[BACKEND] {line.strip()}")
            
    except Exception as e:
        print(f"Error starting backend: {e}")

def main():
    print("Starting Curamete development servers...")
    print("Press Ctrl+C to stop both servers.")
    
    # Start backend in a separate thread
    backend_thread = Thread(target=start_backend)
    backend_thread.daemon = True
    backend_thread.start()
    
    # Give backend a moment to start
    time.sleep(2)
    
    # Start frontend in main thread
    start_frontend()

if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print("\nShutting down servers...")
        sys.exit(0)

File: start-dev.js
// start-dev.js — works from frontend folder where this file lives
import { spawn } from 'child_process';
import { fileURLToPath } from 'url';
import { dirname, resolve } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

console.log('Starting Curamete Development Environment...');
console.log('==========================================\n');

// Resolve repo root (parent of frontend)
const repoRoot = resolve(__dirname, '..'); // D:\curamete\curamete
const backendFolder = resolve(repoRoot, 'backend'); // D:\curamete\curamete\backend
const backendScript = resolve(backendFolder, 'main.py'); // D:\curamete\curamete\backend\main.py

// Spawn backend
console.log('Starting Backend Server...');
const backend = spawn('python', [backendScript], {
  cwd: backendFolder,
  stdio: 'inherit',
  // do NOT set shell:true unless you need shell features; safer without it
});

// If you use Windows and 'python' isn't recognized, try 'py' or full path to python exe.

backend.on('error', (err) => {
  console.error('Error starting backend server:', err);
});

backend.on('close', (code) => {
  console.log(`Backend server exited with code ${code}`);
});

// Start frontend after a short delay so backend boots first
setTimeout(() => {
  console.log('\nStarting Frontend Server...');
  // run `npm run dev` inside the frontend folder (this file's folder)
  const frontend = spawn('npm', ['run', 'dev'], {
    cwd: __dirname,
    stdio: 'inherit',
    // no shell:true
  });

  frontend.on('close', (code) => {
    console.log(`Frontend server exited with code ${code}`);
    // kill backend if still running
    try { backend.kill(); } catch(e){}
    process.exit(code ?? 0);
  });

  frontend.on('error', (error) => {
    console.error('Error starting frontend server:', error);
    try { backend.kill(); } catch(e){}
  });
}, 2000);

// Handle Ctrl+C gracefully
process.on('SIGINT', () => {
  console.log('\nShutting down servers...');
  try { backend.kill(); } catch(e){}
  process.exit(0);
});
